packages/flowtip-react-dom/src/util/state.js

```diff
@@ -0,0 +1,127 @@
+import type {Region} from 'flowtip-core';
+import type {Props, State} from '../types';
+
+/**
+ * Get the last rendered region (`top`, `right`, `bottom`, or `left`).
+ *
+ * @param   {Object} props - FlowTip props.
+ * @param   {Object} state - FlowTip state.
+ * @returns {string} Region.
+ */
+export function getLastRegion(props: Props, state: State): Region | void {
+  return state.result._static === true ? props.region : state.result.region;
+}
+
+/**
+ * Get the current region (`top`, `right`, `bottom`, or `left`) that the FlowTip
+ * layout algorithm should prioritize in it's result.
+ *
+ * @param   {Object} props - FlowTip props.
+ * @param   {Object} state - FlowTip state.
+ * @returns {string} Region.
+ */
+export function getRegion(props: Props, state: State): Region | void {
+  // Feed the current region in as the default if `sticky` is true.
+  // This makes the component stay in its region until it meets a
+  // boundary edge and must change.
+  return props.sticky ? getLastRegion(props, state) : props.region;
+}
+
+/**
+ * Get the dimension of the tail perpendicular to the attached edge of the
+ * content rect.
+ *
+ * Note: `props` are passed in as an argument to allow using this method from
+ * within `componentWillReceiveProps`.
+ *
+ * @param   {Object} props - FlowTip props.
+ * @param   {Object} state - FlowTip state.
+ * @returns {number} Tail length.
+ */
+export function getTailLength(props: Props, state: State): number {
+  const lastRegion = getLastRegion(props, state);
+
+  if (state.tail) {
+    // Swap the width and height into "base" and "length" to create
+    // measurements that are agnostic to tail orientation.
+    if (lastRegion === LEFT || lastRegion === RIGHT) {
+      return state.tail.width;
+    }
+    // Either lastRegion is top or bottom - or it is undefined, which means
+    // the tail was rendered using the static dummy result that uses the
+    // bottom region.
+    return state.tail.height;
+  }
+
+  return 0;
+}
+
+/**
+ * Get the dimension of the tail parallel to the attached edge of the content
+ * rect.
+ *
+ * Note: `props` are passed in as an argument to allow using this method from
+ * within `componentWillReceiveProps`.
+ *
+ * @param   {Object} props - FlowTip props.
+ * @param   {Object} state - FlowTip state.
+ * @returns {number} Tail base size.
+ */
+export function getTailBase(props: Props, state: State): number {
+  const lastRegion = getLastRegion(props, state);
+
+  if (state.tail) {
+    // Swap the width and height into "base" and "length" to create
+    // measurements that are agnostic to tail orientation.
+    if (lastRegion === LEFT || lastRegion === RIGHT) {
+      return state.tail.height;
+    }
+    // Either lastRegion is top or bottom - or it is undefined, which means
+    // the tail was rendered using the static dummy result that uses the
+    // bottom region
+    return state.tail.width;
+  }
+
+  return 0;
+}
+
+/**
+ * Get current minimum linear overlap value.
```

Comment:
I know you've used this term before but is there a definition of what "linear" overlap is vs some other kind of overlap? This comment has little to do with the PR itself, just if it's going to be part of the nomenclature for this library a working definition would be handy. A cursory google reveals no common definition or use of this term.

Reply:
`overlap` is one of the named config options of `flowtip-core`

Reply:
No concern with `overlap` – as above more curious what the `linear` qualifier means.

Reply:
"Linear" here refers to parallel with the edge of the target facing the content, as opposed to perpendicular.

The overlap related computations are effectively 2D with respect to one of the four regions.